server/reload_test.go

```diff
@@ -3808,6 +3818,49 @@ func TestConfigReloadBoolFlags(t *testing.T) {
 			true,
 			func() bool { return opts.TraceVerbose },
 		},
+		// --js override
+		{
+			"jetstream_not_in_config_no_override",
+			"",
+			nil,
+			false,
+			func() bool { return opts.JetStream },
+		},
+		{
+			"jetstream_not_in_config_override_true",
+			"",
+			[]string{"--js"},
+			true,
+			func() bool { return opts.JetStream },
+		},
+		{
+			"jetstream_false_in_config_no_override",
+			"jetstream: false",
+			nil,
+			false,
+			func() bool { return opts.JetStream },
+		},
+		{
+			"jetstream_false_in_config_override_true",
+			"jetstream: false",
+			[]string{"--js"},
+			true,
+			func() bool { return opts.JetStream },
+		},
+		{
+			"jetstream_true_in_config_no_override",
+			"jetstream: true",
+			nil,
+			true,
+			func() bool { return opts.JetStream },
+		},
+		{
+			"jetstream_true_in_config_override_false",
```

Comment:
@wallyqs When adding this (and without the code fix), the test would pass for some reasons, including the fact that Reload() was not doing what it should be doing. This is why I had to add the disable function for the configuration digest.

Arguably, we could have all tests that call Reload() without an actual file change be checked/fixed, but I feel like it is more prudent to disable config digest in most tests but enable only for the ones that specifically check for the digest. Does that sound reasonable?

Reply:
Thanks @kozlovic, I think probably need to revisit the skip reload if no changes to config since I think it would mean that TLS certs changes without modifying the paths to the certs do not trigger a reload.

Reply:
But that was the point of the config digest, no? Meaning don't do the reload if no change is detected. Of course, and I think it may have been mentioned in the original PR, what constitute a change may be difficult when it comes to files.. you may need to do additional checks. Does that mean I should not bother evaluating the tests that call Reload() where nothing happens since you may revisit the config digest?

Reply:
@kozlovic main point of config digest is that we can track in the logs when behavior in the server changed, then I got the idea to use it to skip reload if not needed to avoid redoing imports, etc... but for that I should also keep a digest of the contents of the certs used by the server to confirm whether do need a reload or not, so I think I will disable not doing reload when config is not changed for now